### ISSUE TITLE
Panic due to nil pointer in `Analyze`

### ISSUE THREAD

**Issue by TomerJLevy** [essentialkaos/sslscan#95]
The `Analyze` function crashes due to nil pointer.

Stack Trace:
```
runtime error: invalid memory address or nil pointer dereference
Full Stack Trace
  github.com/essentialkaos/sslscan/v13.(*API).doRequest(0x0, {0xc000f92000, 0x90}, {0x0, 0x0})
    /tmp/go/pkg/mod/github.com/essentialkaos/sslscan/v13@v13.1.1/sslscan.go:650 +0x1da
  github.com/essentialkaos/sslscan/v13.(*API).Analyze(0x0, {0xc000cbb4a0, 0x2a}, {0x0, 0x0, 0x0, 0x0, 0x0})
    /tmp/go/pkg/mod/github.com/essentialkaos/sslscan/v13@v13.1.1/sslscan.go:563 +0x152

```
### Bug report

**System info:**

* **Version used: v13.1.1** 
* **OS (e.g. from `/etc/*-release`): Linux** 
* **Kernel (`uname -a`):** 
* **Go version (`go version`): 1.17** 
* **Install tools: Go** 

**Steps to reproduce:**

Happens sporadically

**Expected behavior:**

Work without crashing

**Actual behavior:**

Crashes

**Additional info:**

That's the API we create (adding a custom DNS resolver):
```
api, err := sslscanv13.NewAPI("SSLScanTester", "10.0.2")
if err != nil || api == nil {
	return nil, err
}

api.RequestTimeout = 5 * time.Minute
api.Client.Dial = (&fasthttp.TCPDialer{
	Concurrency:      4096,
	DNSCacheDuration: time.Hour,
	Resolver: &net.Resolver{
		PreferGo: true,
		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
			d := net.Dialer{
				Timeout: time.Millisecond * time.Duration(10000),
			}
			return d.DialContext(ctx, network, nameServer)
		},
	},
}).Dial
```

And then this crashes, let's say once in ~50 runs:
```
progress, err = api.Analyze(host, sslscanv13.AnalyzeParams{})
```

---

### Feature Request

Opening a feature request kicks off a discussion. Requests may be closed if we're not actively planning to work on them.

**Proposal:**

[Description of the feature]

**Current behavior:**

[What currently happens]

**Desired behavior:**

[What you would like to happen]

**Use case:**

[Why is this important (helps with prioritizing requests)]

**Comment by andyone** [comment]
```go
github.com/essentialkaos/sslscan/v13.(*API).doRequest(0x0, {0xc000f92000, 0x90}, {0x0, 0x0})
    /tmp/go/pkg/mod/github.com/essentialkaos/sslscan/v13@v13.1.1/sslscan.go:650 +0x1da
```

This message points to [this line](https://github.com/essentialkaos/sslscan/blob/v13.1.1/sslscan.go#L650):
```go
	if api.RequestTimeout == 0 {
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
		err = api.Client.Do(req, resp)
	} else {
		err = api.Client.DoTimeout(req, resp, api.RequestTimeout)
	}
```

It means, that your pointer to the API struct somehow becomes nil. Of course, it's a serious bug due to panic and I'll add an additional check for nil, but you have to check other logic of your code for setting struct to nil.

**Comment by andyone** [comment]
@TomerJLevy JFYI I have released version [13.1.3](https://kaos.sh/sslscan/r/v13.1.3) with extra checks for nil.
